internal/monitors/manager.go

```diff
@@ -349,31 +349,31 @@ func (mm *MonitorManager) createAndConfigureNewMonitor(config config.MonitorCust
 
 	// Monitors can add additional extra metrics to allow through such as based on config flags.
 	if monitorExtra := config.GetExtraMetrics(); monitorExtra != nil {
-		extraMetrics = append(extraMetrics, coreConfig.ExtraMetrics...)
 		extraMetrics = append(extraMetrics, monitorExtra...)
-	} else {
-		extraMetrics = coreConfig.ExtraMetrics
 	}
+	extraMetrics = append(extraMetrics, coreConfig.ExtraMetrics...)
 
 	extraMetricsFilter, err := newMetricsFilter(metadata, extraMetrics, coreConfig.ExtraGroups)
 	if err != nil {
 		return fmt.Errorf("unable to construct extraMetrics filter: %s", err)
 	}
 
 	output := &monitorOutput{
-		monitorType:               monitorType,
+		monitorType:               config.MonitorConfigCore().Type,
 		monitorID:                 id,
-		notHostSpecific:           coreConfig.DisableHostDimensions,
-		disableEndpointDimensions: coreConfig.DisableEndpointDimensions,
-		filter:                    coreConfig.Filter,
-		additionalFilter:          extraMetricsFilter,
-		configHash:                configHash,
-		endpoint:                  endpoint,
-		dpChan:                    mm.DPs,
-		eventChan:                 mm.Events,
-		dimPropChan:               mm.DimensionProps,
-		spanChan:                  mm.TraceSpans,
-		extraDims:                 map[string]string{},
+		notHostSpecific:           config.MonitorConfigCore().DisableHostDimensions,
+		disableEndpointDimensions: config.MonitorConfigCore().DisableEndpointDimensions,
+		filterSet: &dpfilters.FilterSet{
+			ExcludeFilters: []dpfilters.DatapointFilter{oldFilter, newFilter},
```

Comment:
I think you could make this line:

```
 			ExcludeFilters: []dpfilters.DatapointFilter{oldFilter, newFilter, dpfilters.Negate(extraMetricsFilter)},
```

And not even need an additional field on the Output instance or the extra logic in the SendDatapoint method.

What do you think?

Reply:
Taking nils into account of course.